--- fft/gf2x-ternary-fft.c	2024-12-09 18:40:28.132018755 +0530
+++ fft/gf2x-ternary-fft.c	2024-12-09 18:45:28.986821869 +0530
@@ -1201,6 +1201,34 @@
     memset(o, 0, sizeof(gf2x_ternary_fft_info_t));
 }
 
+#ifdef _MSC_VER
+int vasprintf(char **strp, const char *format, va_list ap)
+{
+    int len = _vscprintf(format, ap);
+    if (len == -1)
+        return -1;
+    char *str = (char*)malloc((size_t) len + 1);
+    if (!str)
+        return -1;
+    int retval = vsnprintf(str, len + 1, format, ap);
+    if (retval == -1) {
+        free(str);
+        return -1;
+    }
+    *strp = str;
+    return retval;
+}
+
+int asprintf(char **strp, const char *format, ...)
+{
+    va_list ap;
+    va_start(ap, format);
+    int retval = vasprintf(strp, format, ap);
+    va_end(ap);
+    return retval;
+}
+#endif
+
 char * gf2x_ternary_fft_info_explain(gf2x_ternary_fft_info_srcptr p)
 {
     int rc;
